tls/s2n_alerts.c

```diff
@@ -222,7 +222,14 @@ int s2n_process_alert_fragment(struct s2n_connection *conn)
                 conn->config->cache_delete(conn, conn->config->cache_delete_data, conn->session_id, conn->session_id_len);
             }
 
-            /* All other alerts are treated as fatal errors */
+            /*
+             *= https://tools.ietf.org/rfc/rfc8446#section-6
+             *# MUST be treated as error alerts when received
+             *# regardless of the AlertLevel in the message.  Unknown Alert types
+             *# MUST be treated as error alerts.
```

Comment:
A better place for this citation might be in `s2n_process_as_warning()`, since this function checks for < TLS 1.3 before ignoring the alert.

Also, I know the first part of this section isn't related, but this comment is hard to read without it.

I think the sentence about unknown alert types belongs here though, so maybe that part should be separated out.